frontend/test/specs/comment/commentControllerSpec.js

```diff
@@ -0,0 +1,584 @@
+
+        it('should be false when the post is published', function() {
+            commentCtrl.post.state = 'published';
+            expect(commentCtrl.isDeletedPost()).toBeFalsy();
+        });
+    });
+
+    describe('isLikedByUser', function() {
+        it('should be true when the post is liked by the user', function() {
+            commentCtrl.addLike();
+            expect(commentCtrl.isLikedByUser()).toBeTruthy();
+        });
+
+        it('should be false when the post is not liked by the user', function() {
+            commentCtrl.removeLike();
+            expect(commentCtrl.isLikedByUser()).toBeFalsy();
+        });
+    });
+
+    describe('getReplies', function() {
+        it('should get the comment replies', function () {
+            expect(commentCtrl.getReplies()).toEqual([]);
+            commentCtrl.comment.replies[reply.id] = {...reply};
+            // add reply to comment
+            const commentReplies = Object.values(commentCtrl.comment.replies);
+            expect(commentCtrl.getReplies()).toEqual(commentReplies);
+        });
+
+        it('should return an empty list when the comment is a reply', function () {
+            commentCtrl.comment.replies[reply.id] = {...reply};
+            commentCtrl.reply = {...reply};
+            expect(commentCtrl.getReplies()).toEqual([]);
+        });
+    });
+
+    describe('numberOfLikes', function () {
+        it('should return the number of likes of a comment', function () {
+            commentCtrl.comment.likes = [];
+            expect(commentCtrl.numberOfLikes()).toEqual(0);
+            commentCtrl.comment.likes = ['user-key-01', 'user-key-02'];
+            expect(commentCtrl.numberOfLikes()).toEqual(2);
+        });
+
+        it('should return the number of likes of a reply', function () {
+            commentCtrl.reply = {...reply, likes:[]};
+            commentCtrl.setCurrentComment();
+            expect(commentCtrl.numberOfLikes()).toEqual(0);
+            commentCtrl.reply = {
+                ...reply, 
+                likes: ['user-key-01', 'user-key-02']
+            };
+            commentCtrl.setCurrentComment();
+            expect(commentCtrl.numberOfLikes()).toEqual(2);
+        });
+    }); 
+
+    describe('numberOfReplies', function () {
+        it('should return the number of replies of a comment', function() {
+            commentCtrl.comment.replies = {};
+            expect(commentCtrl.numberOfReplies()).toEqual(0);
+            commentCtrl.comment.replies = {1: nilReply, 2: nilReply};
+            expect(commentCtrl.numberOfReplies()).toEqual(2);
+        });
+    });
+
+    describe('canDeleteComment', function () {
+        beforeEach(function() {
+            commentCtrl.comment.likes = [];
+            commentCtrl.comment.replies = {};
+            commentCtrl.comment.author_key = user.key;
+            commentCtrl.post.state = 'published';
+        })
+
+        it('should be true when the post has no activity', function () {
```

Comment:
... when the post has no activity, its not a deleted post and the user is the author